### DISCUSSION TITLE
Livewire::addPersistentMiddleware not working

### DISCUSSION THREAD

**Discussion by eriktobben** [livewire/livewire#4209] [1. Help]
Hi! I have a middleware that checks if the user has is_admin == 1, that is applied on the route. I saw in the docs that you can use Livewire::addPersistentMiddleware, to keep custom middlewares.

I have added this code in the boot method of AppServiceProvider.php:
`Livewire::addPersistentMiddleware([IsAdmin::class]);`

When I change the is_admin to 0 in the database, Livewire still performs the update.

Am I missing something, or have I added the code to wrong ServiceProvider?

**Comment by joshhanley** [comment]
@eriktobben yeah that should work, as long as the middleware is also applied to the route definition, as Livewire inspects the route for all middleware and then filters by the persistent middleware.
